pandas/core/arrays/string_.py

```diff
@@ -208,21 +215,25 @@ def __init__(self, values, copy=False):
         values = extract_array(values)
 
         super().__init__(values, copy=copy)
+        if not isinstance(values, type(self)):
+            self._validate()
         # error: Incompatible types in assignment (expression has type "StringDtype",
         # variable has type "PandasDtype")
         NDArrayBacked.__init__(self, self._ndarray, StringDtype())
-        if not isinstance(values, type(self)):
-            self._validate()
 
     def _validate(self):
         """Validate that we only store NA or strings."""
-        if len(self._ndarray) and not lib.is_string_array(self._ndarray, skipna=True):
-            raise ValueError("StringArray requires a sequence of strings or pandas.NA")
         if self._ndarray.dtype != "object":
             raise ValueError(
                 "StringArray requires a sequence of strings or pandas.NA. Got "
                 f"'{self._ndarray.dtype}' dtype instead."
             )
+        try:
+            lib.ensure_string_array(
```

Comment:
we now always call ensure_string_array when passing a numpy array? how does this affect performance? and how is the constructor now different from _from_sequence

Reply:
xref https://github.com/pandas-dev/pandas/pull/40687/files#r637390830

if the only difference between the constructor and _from_sequence is the coerce argument passed to ensure_string_array, can the coerce parameter be added to _from_sequence instead.

Reply:
> we now always call ensure_string_array when passing a numpy array? how does this affect performance? and how is the constructor now different from _from_sequence

Perf should be unaffected, since we were previously calling ``lib.is_string_array`` everytime and both do a single pass over the data, but I am not sure that the StringArray constructor is covered by asvs. I will check this when I have time.

Responding to your second point, _from_sequence and the constructor differ in that the constructor validates that elements are string/null(raising otherwise), and _from_sequence just casts them.

> xref #40687 (comment)
if the only difference between the constructor and _from_sequence is the coerce argument passed to ensure_string_array, can the coerce parameter be added to _from_sequence instead.

IIUC, _from_sequence is private and I figured that end users of pandas sometimes might like an error to be raised if values contains non-strings(that are also not accepted null values). If not, they would just use pd.array directly(which casts instead of raising).